### PR TITLE
Update Vega plot in notebook

### PR DESCRIPTION
**Pull request by patrick-miller**
So far this PR has made it so the Vega plot doesn't save intermediate files.

I am exploring how to work https://github.com/altair-viz/jupyter_vega into the notebook. I think that will make the plots dynamic in a pre-run notebook.

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            # For floats, convert nan->None: np.float is not JSON serializable
+            col = df[col_name].astype(object)
+            df[col_name] = col.where(col.notnull(), None)
+        elif str(dtype).startswith('datetime'):
+            # Convert datetimes to strings
+            # astype(str) will choose the appropriate resolution
+            df[col_name] = df[col_name].astype(str).replace('NaT', '')
+    return df
+
+def fill_spec_with_data(spec, data):
```
Comment (dhimmel) [inline utils.py:39]: Can we get a docstring?

**Review by dhimmel — COMMENTED** [review]
Great! Just one small doc comment

**Review by dhimmel — APPROVED** [review]

**Comment by dhimmel** [comment]
@patrick-miller before I merge just wanted to make you aware of this GitHub warning:

> Unrecognized author. If this is you, make sure patrick.miller@macmillan.com is associated with your account. Click to add email addresses in your account settings.

In short, `patrick.miller@macmillan.com` isn't associated with your GitHub account. This means the commit will not be associated with your github account. The solution is to:

1. either add `patrick.miller@macmillan.com` as an additional email on GitHub
2. modify your commits to use the email address registered with github or `patrick-miller@users.noreply.github.com`. If using this option, you might as well squash everything, then just change authorship of that one commit.

**Comment by patrick-miller** [comment]
I'll squash everything and change authorship for the commit.

**Comment by dhimmel** [comment]
`2.TCGA-MLexample-covariates.py` should be in `scripts`. Will address later.

**Pull request merged**